### DISCUSSION TITLE
How to use Lightning-Flash with TPU on Kaggle?

### DISCUSSION THREAD

**Discussion by prikmm** [PyTorchLightning/lightning-flash#632] [Q&A] [answered]
I am trying to import flash on kaggle but am constantly experiences bugs. 

pytorch_xla-1.7: `ModuleNotFoundError: No module named 'torch_xla.debug.profiler'`

pytorch_xla-1.8: `ImportError: /opt/conda/lib/python3.7/site-packages/_XLAC.cpython-37m-x86_64-linux-gnu.so: undefined symbol: _ZNK3c104Type14isSubtypeOfExtERKSt10shared_ptrIS0_EPSo`

pytorch_xla-1.8.1: `OSError: /opt/conda/lib/python3.7/site-packages/torchtext/_torchtext.so: undefined symbol: _ZNK3c104Type14isSubtypeOfExtESt10shared_ptrIS0_EPSo`

I want to know how to use on kaggle. Thanks!

PS: using the below given way of installing PyTorch/XLA
```
!curl https://raw.githubusercontent.com/pytorch/xla/master/contrib/scripts/env-setup.py -o pytorch-xla-env-setup.py
!python pytorch-xla-env-setup.py --version 1.8 --apt-packages libomp5 libopenblas-dev
```

Edit: I am installing flash from master-branch

**Comment by akihironitta** [comment, marked as answer]
@prikmm Hi, thanks for opening a discussion! The error may be caused by the wrong pair of versions of PyTorch and xla. Currently, it's tested only with `torch-xla==1.8`, so please use the version and try to match the `torch` version with `torch-xla`. You can check installed versions with `pip list | grep torch`.

Related issue: https://github.com/PyTorchLightning/pytorch-lightning/issues/8315

**Reply by akihironitta** [reply]
Just published a Kaggle notebook. Please have a look at: https://www.kaggle.com/akihironitta/setting-up-lightning-flash-on-tpu

**Reply by prikmm** [reply]
Yup, it's working. Thanks!

**Reply by ethanwharris** [reply]
Awesome!!! Nice work @akihironitta
